### PR TITLE
Split progress into async & http

### PR DESCRIPTION
**Pull request by kenoir**
### What is this PR trying to achieve?

We need to synchronously create and provide IDs for archive jobs.

This PR moves progress -> progress_async, and creates a new progress_http service built around akka-http.

### Who is this change for?

🦂

### CODE REVIEW

**Comment by kenoir** [comment]
@alexwlchan so in retrospect this change set could have been reduced by moving progress -> progress_async separately, and then building progress_async up in a skeleton/happy path/edge case fashion.

I don't think splitting this PR now has much utility as it would be cumbersome to do so (for me at least).

File: archive/archivist/src/test/scala/uk/ac/wellcome/platform/archive/archivist/ArchivistFeatureTest.scala [hunk 1]
```diff
@@ -305,15 +302,15 @@ class ArchivistFeatureTest
...
+                            Some(callbackUri)
                           ),
                           ArchiveComplete(
                             requestId2,
                             BagLocation(
                               storageBucket.name,
                               "archive",
                               BagPath(s"$DigitisedStorageType/$validBag2")),
-                            Some(callbackUrl)
+                            Some(callbackUri)
```
[the remark this answers is not in the document]
Comment (kenoir) [reply archive/archivist/src/test/scala/uk/ac/wellcome/platform/archive/archivist/ArchivistFeatureTest.scala]: This isn't just a name change, the issue here is different services using strings/URIs to represent resource addressed.

File: archive/common/src/main/scala/uk/ac/wellcome/platform/archive/common/models/CallbackNotification.scala [hunk 2]
```diff
@@ -0,0 +1,21 @@
+package uk.ac.wellcome.platform.archive.common.models
+
+import uk.ac.wellcome.platform.archive.common.progress.models.Progress
+
+case class CallbackNotification(id: String,
+                                callbackUrl: String,
```
[the remark this answers is not in the document]
Comment (kenoir) [reply archive/common/src/main/scala/uk/ac/wellcome/platform/archive/common/models/CallbackNotification.scala]: One's a string, the other a URI - they should both be URIs, but probably not in the PR.

File: archive/common/src/main/scala/uk/ac/wellcome/platform/archive/common/progress/monitor/ProgressMonitor.scala [hunk 3]
```diff
@@ -28,6 +29,21 @@ class ProgressMonitor(
       DateTimeFormatter.ISO_INSTANT.format(_)
     )
 
+  def get(id: String) = {
+    Scanamo.get[Progress](dynamoClient)(dynamoConfig.table)(
+      'id -> id
+    ) match {
+      case Some(Right(progress)) => Some(progress)
+      case Some(Left(error)) => {
+        val exception = new RuntimeException(
```
[the remark this answers is not in the document]
Comment (kenoir) [reply archive/common/src/main/scala/uk/ac/wellcome/platform/archive/common/progress/monitor/ProgressMonitor.scala]: I think we can leave that to a consumer (or create a more specifically typed error here - though I think it remains a RuntimeException).

File: archive/progress_http/src/main/resources/application.conf [hunk 4]
```diff
@@ -0,0 +1,161 @@
+#####################################
```
[the remark this answers is not in the document]
Comment (kenoir) [reply archive/progress_http/src/main/resources/application.conf]: I prefer here that the config is explicit although it may be the default.

File: archive/progress_http/src/main/resources/logback.xml [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+  </appender>
+
+  <root level="${log_level:-INFO}">
+    <appender-ref ref="STDOUT" />
+  </root>
+
+  <!-- reduce external logging -->
+  <logger name="org.apache.http" level="ERROR"/>
+  <logger name="io.netty" level="ERROR"/>
+  <logger name="com.twitter" level="WARN"/>
```
[the remark this answers is not in the document]
Comment (kenoir) [reply archive/progress_http/src/main/resources/logback.xml]: Probably, not.

File: archive/progress_http/src/main/resources/application.conf (hunk 4 above)
[the remark this answers is not in the document]
Comment (alexwlchan) [reply archive/progress_http/src/main/resources/application.conf]: How do you tell when an application deviates from the default?

[the remark this answers is not in the document]
Comment (kenoir) [reply archive/progress_http/src/main/resources/application.conf]: Hmmm, good point - as I don't know where it does now! I'll try and remove these config files and see what happens.

**Comment by kenoir** [comment]
Part of: https://github.com/wellcometrust/platform/issues/2634
